packages/wallets/src/solana/wallet.ts

```diff
@@ -1,7 +1,13 @@
 import type { Connection, PublicKey, Transaction, VersionedTransaction } from "@solana/web3.js";
 
+import type { CreateWalletDto } from "@/api/gen/types.gen";
+
 import type { ApiClient } from "../api";
 
+export type SolanaSigner = NonNullable<
+    Extract<CreateWalletDto, { type: "solana-smart-wallet" }>["config"]
+>["adminSigner"];
+
```

Comment:
Moved the signer definition into the module itself, similar to EVM.

@alberto-crossmint I've tried to minimize Solana-related changes, but I think this one is necessary. You'll probably want to use this to instantiate the `SolanaWallet` anyway.